app/controllers/account/permissions_controller.rb

```diff
@@ -13,15 +13,46 @@ def show
 
   def edit
     authorize [:account, @application], :edit_permissions?
+
+    @shared_permissions_form_locals = {
+      action: account_application_permissions_path(@application),
+      application: @application,
+      cancel_path: account_applications_path,
+      user: current_user,
+    }
+
+    if @permissions.count > 8
+      @current_permissions = []
+      @new_permission_options = []
+      @permissions.each do |permission|
+        if current_user.has_permission?(permission)
+          @current_permissions.push(permission)
+        else
+          @new_permission_options.push({ text: permission.name, value: permission.id })
+        end
+      end
+    end
   end
 
   def update
     authorize [:account, @application], :edit_permissions?
 
+    selected_permission_ids = []
+
+    if update_params[:supported_permission_ids]
+      selected_permission_ids = update_params[:supported_permission_ids]
+    elsif update_params[:new_permission_id]&.length&.> 0
```

Comment:
Would `length` here ever return `nil`? Can we remove the `&. > 0` or am I misunderstanding here?

Reply:
I've found this condition a little tricky to read, though I know it's hard as I've spent half the day trying to come up with an alternative 😅...

I don't know if this is any clearer, I initially tried an early return but couldn't get it to work:

```
    selected_permission_ids = update_params[:supported_permission_ids] || []

    if selected_permission_ids.empty?
      if update_params[:new_permission_id]&.length > 0
        selected_permission_ids.concat(update_params[:current_permission_ids] || [], [update_params[:new_permission_id]])
      else
        flash[:alert] = "You must select a permission."
        redirect_to edit_account_application_permissions_path(@application)
        return
      end
    end

```